src/garage/sampler/base.py

```diff
@@ -1,43 +1,31 @@
+"""Base class of Sampler."""
 import abc
 
 
 class Sampler(abc.ABC):
+    """Base class for sampler."""
+
     @abc.abstractmethod
     def start_worker(self):
-        """Initialize the sampler,
+        """Initialize the sampler.
 
         e.g. launching parallel workers if necessary.
         """
-        pass
 
     @abc.abstractmethod
-    def obtain_samples(self, itr):
+    def obtain_samples(self, itr, batch_size, whole_paths):
         """Collect samples for the given iteration number.
 
         Args:
-            itr (int): Iteration number.
+            itr (int): number of iteration
+            batch_size (int): number of batch size
+            whole_paths (bool): whether to use whole path or truncated
 
         Returns:
             list[dict]: A list of paths.
 
         """
-        pass
 
     @abc.abstractmethod
     def shutdown_worker(self):
         """Terminate workers if necessary."""
-        pass
-
-
-class BaseSampler(Sampler):
```

Comment:
why delete this? it was created as part of a normal pattern: defining an interface (`Sampler`) distinct from its most common implementation `BaseSampler`, even of the base implementation is obvious.